tools/converter/common/src/main/java/org/wso2/ei/tools/converter/common/ballerinahelper/Message.java

```diff
@@ -81,26 +81,43 @@ public static void getPayload(BallerinaASTModelBuilder ballerinaASTModelBuilder,
     /**
      * Set the payload of a message
      */
-    public static void setPayload(BallerinaASTModelBuilder ballerinaASTModelBuilder, Map<String, Object> parameters) {
+    public static void setPayload(BallerinaASTModelBuilder ballerinaASTModelBuilder, Map<String, Object> parameters,
+            boolean isVariableCreationNeeded) {
 
         if (Constant.JSON.equals((String) parameters.get(Constant.TYPE))) {
-            ballerinaASTModelBuilder.addComment(Constant.BLANG_COMMENT_JSON);
-            ballerinaASTModelBuilder.addTypes(Constant.BLANG_TYPE_JSON); //type of the variable
-            ballerinaASTModelBuilder.createStringLiteral((String) parameters.get(Constant.FORMAT));
-            ballerinaASTModelBuilder.createVariable((String) parameters.get(Constant.PAYLOAD_VAR_NAME), true); //name of
-            // the variable
+            if (isVariableCreationNeeded) {
+                ballerinaASTModelBuilder.addComment(Constant.BLANG_COMMENT_JSON);
+                ballerinaASTModelBuilder.addTypes(Constant.BLANG_TYPE_JSON); //type of the variable
+                ballerinaASTModelBuilder.createStringLiteral((String) parameters.get(Constant.FORMAT));
+                ballerinaASTModelBuilder
+                        .createVariable((String) parameters.get(Constant.PAYLOAD_VAR_NAME), true); //name of
+                // the variable
+            }
             ballerinaASTModelBuilder.createNameReference(Constant.BLANG_PKG_MESSAGES, Constant.BLANG_SET_JSON_PAYLOAD);
 
         } else if (Constant.XML.equals((String) parameters.get(Constant.TYPE))) {
-            ballerinaASTModelBuilder.addTypes(Constant.BLANG_TYPE_XML); //type of the variable
-            ballerinaASTModelBuilder.addComment("//TODO: Change the double quotes to back tick. ");
-            ballerinaASTModelBuilder.createXMLLiteral((String) parameters.get(Constant.FORMAT));
-            ballerinaASTModelBuilder.createVariable((String) parameters.get(Constant.PAYLOAD_VAR_NAME), true); //name of
-            // the variable
+            if (isVariableCreationNeeded) {
+                ballerinaASTModelBuilder.addTypes(Constant.BLANG_TYPE_XML); //type of the variable
+                ballerinaASTModelBuilder.addComment("//TODO: Change the double quotes to back tick. ");
```

Comment:
This is a comment in generated ballerina code. Code changes that need to be done before using the generated code is injected as TODO tasks in generated code.